src/frontend/screens/PhotoPreviewModal/index.tsx

```diff
@@ -0,0 +1,497 @@
+  },
+});
+
+export function PhotoPreviewModal({
+  route,
+}: NativeRootNavigationProps<'PhotoPreviewModal'>) {
+  const {observationDocId, photo} = route.params;
+  const {projectId} = useActiveProject();
+  const {formatMessage: t, formatDate, formatTime} = useIntl();
+
+  const coordinateFormat = useCoordinateFormat();
+
+  // TODO: This check needs to be updated in the case of the saved photo using attachment fields that will
+  // be available in an unreleased version of core.
+  const isValidatedByCoMapeo =
+    photo.type === 'photo'
+      ? true
+      : typeof photo.mediaMetadata.timestamp === 'number' &&
+        photo.mediaMetadata;
+
+  const exif =
+    photo.type === 'photo' ? photo.photoExif : photo.mediaMetadata.photoExif;
+
+  const [imageStorageSize, setImageStorageSize] = useState<number | undefined>(
+    undefined,
+  );
+
+  // TODO: For saved photo, use attachment's `createdAt` field (available in unreleased version of core)
+  const timestamp =
+    photo.type === 'processed' ? photo.mediaMetadata.timestamp : undefined;
+
+  const coordinates =
+    photo.type === 'processed' && photo.mediaMetadata.location
+      ? {
+          lon: photo.mediaMetadata.location.coords.longitude,
+          lat: photo.mediaMetadata.location.coords.latitude,
+        }
+      : photo.type === 'photo'
+        ? // TODO: Use attachment's `position` field (not yet implemented in schema)
+          undefined
+        : undefined;
+
+  const deviceDetails = exif
+    ? getDeviceDetailsText({make: exif.Make, model: exif.Model})
+    : null;
+
+  const cameraDetails = exif
+    ? getCameraDetailsText(
+        {
+          // @ts-expect-error Need to update schema
+          fNumber: exif.FNumber,
+          orientation: exif.Orientation,
+        },
+        t,
+      )
+    : null;
+
+  const photoDetails =
+    typeof exif?.ImageWidth === 'number' &&
+    typeof exif?.ImageLength === 'number'
+      ? getPhotoDetailsText(
+          {
+            width: exif.ImageWidth,
+            height: exif.ImageLength,
+            storageSize: imageStorageSize,
+          },
+          t,
+        )
+      : null;
+
+  return (
+    <ScrollView contentContainerStyle={{padding: 20, gap: 20}}>
+      <View>
+        {photo.type === 'photo' ? (
+          <Suspense fallback={null}>
+            <View style={{flex: 1, borderRadius: 10, overflow: 'hidden'}}>
+              <ErrorBoundary fallback={() => <ImageErrorPlaceholder />}>
+                <CoreBlobImage
+                  driveId={photo.driveDiscoveryId}
+                  name={photo.name}
+                  projectId={projectId}
+                  onLoad={event => {
+                    getExpoImageStorageSize(event.source.url)
+                      .then(size => {
+                        setImageStorageSize(size);
+                      })
+                      .catch(err => {
+                        captureException(err);
+                      });
+                  }}
```

Comment:
similar comment about why we're doing this via `onLoad` instead of alternative approaches.